compiler-cli/src/dependencies.rs

```diff
@@ -518,6 +522,49 @@ fn get_manifest<Telem: Telemetry>(
     }
 }
 
+fn is_same_requirements(
+    requirements1: HashMap<EcoString, Requirement>,
+    requirements2: HashMap<EcoString, Requirement>,
+    root_path: &Utf8Path,
+) -> Result<bool, gleam_core::Error> {
+    let canonicalised_requirements2: HashMap<EcoString, Requirement> = requirements2
+        .into_iter()
+        .map(|(string, requirement)| {
+            Ok((
+                string,
+                match requirement {
+                    Requirement::Hex { .. } => requirement,
+                    Requirement::Path { path } => Requirement::Path {
+                        path: fs::canonicalise(&root_path.join(path))?,
+                    },
+                    Requirement::Git { .. } => requirement,
+                },
+            ))
+        })
+        .collect::<Result<Vec<_>, gleam_core::Error>>()?
+        .into_iter()
+        .collect();
+
+    let canonicalised_requirements1: HashMap<EcoString, Requirement> = requirements1
+        .into_iter()
+        .map(|(string, requirement)| {
+            Ok((
+                string,
+                match requirement {
+                    Requirement::Hex { .. } => requirement,
+                    Requirement::Path { path } => Requirement::Path {
+                        path: fs::canonicalise(&root_path.join(path))?,
+                    },
+                    Requirement::Git { .. } => requirement,
+                },
+            ))
+        })
+        .collect::<Result<Vec<_>, gleam_core::Error>>()?
+        .into_iter()
+        .collect();
+    Ok(canonicalised_requirements1 == canonicalised_requirements2)
```

Comment:
This function could have the duplication removed and run faster if instead of making two new hashmaps it checks that they are the same size, and then iterates over one and pulls the same key from the other, checking if they match as appropriate.